src/DynamoCore/Engine/Profiling/NodeProfilingData.cs

```diff
@@ -11,6 +11,8 @@ class NodeProfilingData : IDisposable
         private DateTime? startTime = null;
```

Comment:
being `startTime` and `endTime` nullable properties, wouldn't it be easier to use them to check if profiling has or not started? Still you'll need to reset them values to `null` before each run, but cleaner that having to deal with an extra property.

Reply:
@alvpickmans yeah that would work. One effect of this approach is that the `ExecutionTimeData` would only contain results for nodes executed during the most recent graph run. All other results would be null, so it would be up to the developer to cache previous results. I think I actually like this because it makes it obvious which results are "fresh".